pages/_app.js

```diff
@@ -3,12 +3,30 @@ import '../styles/globals.css';
 import { Header, GithubCodeLink } from '../components';
 import SEO from '@bradgarropy/next-seo';
 import { motion } from 'framer-motion';
+
+import { useEffect, useState } from 'react';
+
 import NextNProgress from 'nextjs-progressbar';
 
 function MyApp({ Component, pageProps, router }) {
+
+  const [favicon, setFavicon] = useState("/faviconLight.ico");
+  const changeFavicon = () => {
+    const isDark = window.matchMedia("(prefers-color-scheme: dark)");
+    if (!isDark.matches)
+      return setFavicon("/faviconDark.ico");
+    return setFavicon("/faviconLight.ico");
+  };
+
+  useEffect(() => {
+    changeFavicon(); 
+    window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', e => changeFavicon());
```

Comment:
Everything else looks good to me just that in the code, you can change the arrow function to just reference `changeFavicon` directly

`window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', changeFavicon);`